src/services/dealer.ts

```diff
@@ -224,3 +224,31 @@ export const requestMatelsoIntegration = async ({
     return handleValidationError(error, { swallowErrors: true })
   }
 }
+
+export const requestWhatsAppIntegration = async ({
+  dealerId,
+  whatsAppNumber,
+  options = {},
+}: {
+  dealerId: number
+  whatsAppNumber: string
+  options?: ApiCallOptions
+}): Promise<WithValidationError> => {
+  try {
+    const result = await postData({
+      path: `dealers/${dealerId}/request-whatsapp-integration`,
+      body: { whatsAppNumber },
+      options: {
+        isAuthorizedRequest: true,
+        ...options,
+      },
+    })
+
+    return {
+      tag: "success",
+      result,
+    }
+  } catch (error) {
+    return handleValidationError(error, { swallowErrors: true })
```

Comment:
can you help me understand - what errors do we swallow and why?
what validation errors could occur in this case?

Reply:
If you add `{ swallowErrors: true }` response errors other than 400 and 422 (as well as other exceptions) will be returned as:
```
{
  tag: "error",
  message: "validation.other-error",
  errors: [],
}
```

that's how we typically handle non-critical API requests
(check the code for reference: https://github.com/carforyou/carforyou-api-client-pkg/blob/master/src/base.ts#L192)

Reply:
1. I used the same code (copy/paste) and I didnt check what that part does, thanks @Averethel 
2. we send the whatsappNumber, and we get the validation error if not populated

Reply:
ah, so this is going to be used in a form... thanks for the explanation, makes sense